3993

MohitKumar020291 opened the pull request:
Fixes #3993 

Changes made in this Pull Request: I had changed all `RMSD.run()` to `RMSD.timeseries()` . I did not find any `run()` method in `msd.py`. I also changed in `test_msd.py` from `.run()`  to `.timeseries()` as per the instructions in issue #3993 
 - 


PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

MohitKumar020291 committed:
Trying fixing some changes for _run() to .timeseries

MohitKumar020291 committed:
Changes in MSD_test

MohitKumar020291 committed:
Changd _run() to .timeseries()

hmacdope commented:
Hi @MohitKumar020291, thanks for the start of this contribution. 
This is off the mark a little because of some confusion round the issue which I could have made clearer. We are talking about replacing the **contents** of the `_run` method with the `Reader.timeseries` method **NOT** the `Results.timeseries` attribute.

hmacdope commented on benchmarks/benchmarks/analysis/rms.py:
```diff
@@ -72,10 +72,10 @@ def setup(self, select, weights):
 
     def time_RMSD(self, select, weights):
-        """Benchmark RMSD.run() method, which parses
+        """Benchmark RMSD.timeseries() method, which parses
```
This is not correct, you need to `run()` the analysis method.

hmacdope commented on testsuite/MDAnalysisTests/analysis/test_msd.py:
```diff
@@ -61,7 +61,7 @@ def random_walk_u():
 def msd(u, SELECTION):
     # non fft msd
     m = MSD(u, SELECTION, msd_type='xyz', fft=False)
-    m.run()
```
Change this back.

hmacdope commented on testsuite/MDAnalysisTests/analysis/test_msd.py:
```diff
@@ -80,7 +80,7 @@ def test_notidynamics(u, SELECTION):
         u = mda.Universe(PSF, DCD)
         msd = MSD(u, SELECTION)
-        msd.run()
+        msd.timeseries()
```
Change this back.

hmacdope commented on testsuite/MDAnalysisTests/analysis/test_msd.py:
```diff
@@ -121,7 +121,7 @@ def test_simple_step_traj_all_dims(self, step_traj, NSTEP, dim,
         # testing the "simple" algorithm on constant velocity trajectory
         # should fit the polynomial y=dim_factor*x**2
         m_simple = MSD(step_traj, 'all', msd_type=dim, fft=False)
-        m_simple.run()
```
Still have to `run()` change this back.

hmacdope commented on testsuite/MDAnalysisTests/analysis/test_msd.py:
```diff
@@ -134,7 +134,7 @@ def test_simple_start_stop_step_all_dims(self, step_traj, NSTEP, dim,
         # testing the "simple" algorithm on constant velocity trajectory
         # test start stop step is working correctly
         m_simple = MSD(step_traj, 'all', msd_type=dim, fft=False)
-        m_simple.run(start=10, stop=1000, step=10)
```
same here.

hmacdope reviewed (changes requested):
You need to change the actual contents of the `_run` method in `analysis/msd.py` rather than the invocations of the MSD analysis.

MohitKumar020291 commented:
@hmacdope Are other changes working well?

hmacdope commented:
No, the other changes of run to time series are also not correct, although I understand your confusion. Try and have a look at the _run method in msd.py.:)

MohitKumar020291 commented:
@hmacdope thanks for guiding me. I gone through msd.py but did not found `_run`. I also have two questions . First, that should I change ( `results` to `readers` ) in `AnalysisBase(object)` from base.py ?But I think it will cause unexpected results and will need to do same changes in  `__init__` of base.py which will lead to change in whole code base. Second is should I change **all**   `results` to `reader` in msd.py.

hmacdope commented:
Sorry @MohitKumar020291 i mean_single_frame no run.

MohitKumar020291 committed:
created a new function _single_frame_byReader in msd.py

orbeckst commented:
@MohitKumar020291 some relevant background on analysis classes in MDA: https://userguide.mdanalysis.org/stable/examples/analysis/custom_trajectory_analysis.html

Also note that the tests ("checks") need to pass.

MohitKumar020291 commented:
@orbeckst So can I create a new  function like `_single_frame`  in base.py(or at other place) which will be work for msd.py to run() and I am also confused about if I will change to new function like `_single_frame` , will it work in test_msd.py . I know I am asking a lot , sorry for that.

MohitKumar020291 committed:
Update msd.py

Removed `single_frame` from msd.py and set `_ position_array` in function `_prepare`.

orbeckst commented on package/MDAnalysis/analysis/msd.py:
```diff
@@ -365,16 +366,7 @@ def _parse_msd_type(self):
 
         self.dim_fac = len(self._dim)
-
-    def _single_frame(self):
```
removing _single_frame is the correct approach

orbeckst commented on package/MDAnalysis/analysis/msd.py:
```diff
@@ -346,7 +346,8 @@ def _prepare(self):
                                                   self.n_particles))
         self._position_array = np.zeros(
             (self.n_frames, self.n_particles, self.dim_fac))
-        # self.results.timeseries not set here
+        # set the position array here to remove _single_frame from msd.py 
+        self.results.timeseries = self._position_array
```
Obtain the coordinates via [trajectory.timeseries](https://docs.mdanalysis.org/stable/documentation_pages/coordinates/base.html#MDAnalysis.coordinates.base.ReaderBase.timeseries).

orbeckst reviewed (changes requested):
You made some progress! Removing `_single_frame()` is good but you haven't provided the coordinates yet that are needed in the `_conclude`.

You should use the `timeseries` method of a coordinate reader to extract the coordinates into a numpy array. Putting it into `_prepare` is sensible but you need to make sure you use the `start`, `stop`, `step` arguments that are provided to `run()` (read more about using AnalysisBase).

You'll also need to add yourself to AUTHORS and make an entry in CHANGELOG. Please also introduce yourself on the developer list (see https://github.com/MDAnalysis/mdanalysis/pull/4004#pullrequestreview-1264653043).

Note that after your changes, the tests should still pass.

MohitKumar020291 committed:
Updated msd.py

Implemented `timeseries` with numpy array( as timeseries extracts the cooordinates ). 
Thanks for your guidance.

MohitKumar020291 committed:
Updated msd.py

MohitKumar020291 committed:
Updated msd.py

MohitKumar020291 committed:
Updated msd.py

MohitKumar020291 committed:
updated msd.py

MohitKumar020291 committed:
Updated msd.py

MohitKumar020291 committed:
updated msd.py

MohitKumar020291 committed:
Updated msd.py

MohitKumar020291 committed:
Update msd.py

MohitKumar020291 committed:
Update msd.py

MohitKumar020291 committed:
Update msd.py

MohitKumar020291 committed:
Update msd.py

orbeckst commented:
@MohitKumar020291 , are you editing files via the GitHub webinterface? I am asking because your commit message is always the same and the it is the one that the webinterface prepopulates. *If* you are using the webinterface, I suggest you instead work with a local copy (see the Contribution Guide) and learn to edit and test locally as this leads to much faster and thus  more efficient edit/test cycles. It's also the only sensible way to do multiple edits in different files. (If you're already working locally and just occasionally `git push` your changes then you can ignore my advice.)

MohitKumar020291 commented:
@orbeckst Yes was editing file via webinterface. Will use local now .Also I read for finding the way to create numpy array in `_prepare`  using coordinate reader but could not get the actuall way to insert start stop step. And I also thinks that there is no need to `_conclude`

MohitKumar020291 committed:
Update msd.py

collected coordinates in _prepare and passed to _conclude

MohitKumar020291 committed:
KatGya

MohitKumar020291 committed:
firkatgya

MohitKumar020291 committed:
Update msd.py

MohitKumar020291 committed:
Update msd.py

MohitKumar020291 committed:
added __init__

MohitKumar020291 committed:
msd.py

MohitKumar020291 committed:
Merge branch '3993' of https://github.com/MohitKumar020291/mdanalysis into 3993

MohitKumar020291 committed:
defining start,stop,step same as test_msd.py

MohitKumar020291 committed:
Fixed u attribute erro

orbeckst reviewed (changes requested):
Your latest changes make very little sense to me.

I know that you have spent some time trying to make this work but I think you're lacking a fundamental understanding how the MDAnalysis analysis functions work. I suggest you close this PR and work on an easier problem first.

MohitKumar020291 commented:
@orbeckst Thanks for advice , I had given a lot of time would like to try last 1 - 2 times.

orbeckst commented:
Ok, my advice is just to first try to understand what you’re attempting to do. This could mean looking at existing code. For instance, understanding what happens when the analysis class is instanced and when the run() method is called.

SophiaRuan commented:
> @orbeckst Thanks for advice , I had given a lot of time would like to try last 1 - 2 times.

Hi @MohitKumar020291, I am also working on this issue and didn't realize that there is supposed to be only one person working on one issue. Please let me know if you want to keep trying to solve this issue. Thank you and good luck!

MohitKumar020291 commented:
@SophiaRuan Yes will continue working on this issue for some next days at least. Sorry I did not understood your "supposed" sentence , is it something like working together on this issue? Thanks.

SophiaRuan commented:
> @SophiaRuan Yes will continue working on this issue for some next days at least. Sorry I did not understood your "supposed" sentence , is it something like working together on this issue? Thanks.

@MohitKumar020291 I mean if you keep working on this issue, I will not continue working on it as there should be only 1 person working on each issue.

MohitKumar020291 commented:
@SophiaRuan Sorry for your inconvenience , I will give some more try.

MohitKumar020291 commented:
@orbeckst and @SophiaRuan I am closing this PR . Thanks.

orbeckst commented:
Thank you @MohitKumar020291 for your efforts.

MohitKumar020291 commented:
@orbeckst Thanks for your gratitude, will continue working on this repo need to clear some more basics, you guys are great mentors I had ever met on github.

RMeli commented:
> need to clear some more basics

The [MDAnalysis User Guide](https://userguide.mdanalysis.org/stable/index.html#) is a fantastic place to learn more about MDAnalysis. I's suggest to spend some time working through the examples.
